### PR TITLE
[FrameworkBundle] allow enabling the HTTP cache using semantic configuration

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Right now, using the HTTP cache requires tweaking the `public/index.php` file [as explained in the doc](https://symfony.com/doc/current/http_cache.html).

This PR removes this requirement by allowing one to do this instead:
```yaml
framework:
    http_cache: true
```

### CODE REVIEW

**Comment by javiereguiluz** [comment]
This looks nice 👍  Thanks for the config example in the description. This will help the Doc team when this feature has to be documented.

**Review by yceruto — APPROVED** [review]
Nice idea, thanks !

**Comment by nicolas-grekas** [comment]
PR is ready.

**Comment by maxhelias** [comment]
Nice! I love the idea 👍

**Comment by nicolas-grekas** [comment]
Since creating HttpCache outside of the framework was done for perf purpose, I did some benchmarks.
Here are my numbers:

 | Enabling point | index.php | framework.yaml
| ------------- | --- | ---
 | Without preloading | 2150 r/s | 1680 r/s
 | Within preloading | 2400 r/s | 2315 r/s

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -180,6 +181,36 @@ private function addFormSection(ArrayNodeDefinition $rootNode)
...
+                        ->booleanNode('debug')->defaultValue('%kernel.debug%')->end()
+                        ->enumNode('trace_level')
+                            ->values(['none', 'short', 'full'])
+                        ->end()
+                        ->scalarNode('trace_header')->end()
+                        ->integerNode('default_ttl')->end()
+                        ->arrayNode('private_headers')
+                            ->performNoDeepMerging()
+                            ->requiresAtLeastOneElement()
+                            ->fixXmlConfig('private_header')
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php:201]: this must be on the parent node to work

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -180,6 +181,36 @@ private function addFormSection(ArrayNodeDefinition $rootNode)
...
+                    ->children()
+                        ->booleanNode('debug')->defaultValue('%kernel.debug%')->end()
+                        ->enumNode('trace_level')
+                            ->values(['none', 'short', 'full'])
+                        ->end()
+                        ->scalarNode('trace_header')->end()
+                        ->integerNode('default_ttl')->end()
+                        ->arrayNode('private_headers')
+                            ->performNoDeepMerging()
+                            ->requiresAtLeastOneElement()
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php:200]: why would we require at least one element here ?

**Pull request merged**